### ISSUE TITLE
Suppressing source description not working

### ISSUE THREAD

**Issue by juliacmcf** [diamm/diamm#445]
I wanted to suppress the publication of the information about the Abbotsbury Rotulus https://www.diamm.ac.uk/sources/4750/#/ while we're working on it (and trying to disentangle Meg's notes), so I didn't click in the checkbox for 'Source description is public'. However, it's showing on the main site despite this, and the only way we could now remove it would be to delete all the work we've already done!

**Comment by ahankinson** [comment]
You could move it to a private note?

**Comment by juliacmcf** [comment]
Not all the inventory and other stuff that we’re working on in multiple fields. Should it only suppress the description field then? It would be preferable to suppress the entire source — I assumed that’s what it would do. Can that be done if we have a source that we don’t want to show until we’ve had time to do all the construction work?

**Comment by ahankinson** [comment]
There are currently 32 sources with their source description marked as "not public" 

I can implement the functionality for suppressing them, but these 32 should be reviewed first before I do to make sure that I don't inadvertently make something disappear.

@fattoxxon is this something you could do? Just check the box if they're supposed to be public. 

- https://www.diamm.ac.uk/sources/4853
- https://www.diamm.ac.uk/sources/4969
- https://www.diamm.ac.uk/sources/4788
- https://www.diamm.ac.uk/sources/4687
- https://www.diamm.ac.uk/sources/4758
- https://www.diamm.ac.uk/sources/4902
- https://www.diamm.ac.uk/sources/4795
- https://www.diamm.ac.uk/sources/4945
- https://www.diamm.ac.uk/sources/4854
- https://www.diamm.ac.uk/sources/4791
- https://www.diamm.ac.uk/sources/4944
- https://www.diamm.ac.uk/sources/4856
- https://www.diamm.ac.uk/sources/5447
- https://www.diamm.ac.uk/sources/4850
- https://www.diamm.ac.uk/sources/5088
- https://www.diamm.ac.uk/sources/4966
- https://www.diamm.ac.uk/sources/4924
- https://www.diamm.ac.uk/sources/4681
- https://www.diamm.ac.uk/sources/4237
- https://www.diamm.ac.uk/sources/4855
- https://www.diamm.ac.uk/sources/4983
- https://www.diamm.ac.uk/sources/4852
- https://www.diamm.ac.uk/sources/4948
- https://www.diamm.ac.uk/sources/4968
- https://www.diamm.ac.uk/sources/4946
- https://www.diamm.ac.uk/sources/5446
- https://www.diamm.ac.uk/sources/4970
- https://www.diamm.ac.uk/sources/4947
- https://www.diamm.ac.uk/sources/4792
- https://www.diamm.ac.uk/sources/4832
- https://www.diamm.ac.uk/sources/4794
- https://www.diamm.ac.uk/sources/4790

**Comment by juliacmcf** [comment]
Previously, checking them as not public left them public, which was a bit frustrating, as we didn’t want pages under construction to show, particularly if we weren’t entirely sure about the listing. I’m surprised there are that many, but they may be ones David is working on?

**Comment by fattoxxon** [comment]
Done

Most were mistakenly left not public - that is, the box just didn't get
ticked & the default is 'not public'. Among the 32 are several of Richard
D's fragments, a couple of Dominique Gatte's ones, and plenty of other
recent fragment finds; also some sources  I know were added recently (the
San Petronio updates, the other partbooks in the Q22 set) and likewise
meant to be public.  I've ticked all those as public.

Also no public are the Sadler reconstructions: I imagine you had intended
those to be public though? Haven't ticked them yet pending confirmation.

Also not public are a handful of those 'virtual reconstruction' mss which
haven't actually had anything done with them yet - just placeholders for
future work. I've left those 'not public'

David

**Comment by ahankinson** [comment]
OK, I think I've fixed this. 

If you are logged in and have a "staff" account, you will see all sources, including those that are not "public". I did this because I thought editors would like to see how the record looks while they're making it.

If you are logged in with a regular account, or not logged in at all, you'll get a "not found" error.

I've tried to find all the places where it may have "leaked" -- for example, in a listing of the sources in an archive, an extra check needs to be done to make sure the MS isn't listed at all to those who aren't authorized, otherwise we get a link to a source that is missing. 

But if you spot any other places, please let me know.

**Comment by fattoxxon** [comment]
Yes, the Sadlers were supposed to be public and I’;m sure we had ticked them as public as someone was asking about them ages ago. There was definitely an issue some while back that if you clicked ’not public’ the source still showed up anyway, so we may not have bothered unchecking that box.

**Comment by juliacmcf** [comment]
AHA!!!
